src/spotishush.js

```diff
@@ -204,6 +218,51 @@ function idagioSetupAdsObserver (progressBar) {
   return mo
 }
 
+function youtubeSetupAdsObserver (adContainer) {
+
+  const mo = new MutationObserver(mutations => {
+    LOG('Mutations: ', mutations)
+
+    for (const mutation of mutations) {
+      for (const addedNode of mutation.addedNodes) {
+        if (addedNode.classList.contains('ytp-ad-player-overlay')) {
+          LOG('Ad detected in our song queue, muting tab...')
+          sendToBg({ action: 'mute' })
+	        return
+        } 
+      }
+ 
+    for (const removedNode of mutation.removedNodes) {
+	    LOG('Node removed', removedNode)
+      if (removedNode.classList.contains('ytp-ad-player-overlay')) {
+	      sendToBg({ action: 'unmute' })
+	      return
+      }
+
+      }
+    }
+
+  }) 
+
+// youtubeHandleAd(mutations[0].target))
+  mo.observe(adContainer, {
+    childList: true
+  })
+
+  youtubeHandleAd(adContainer)
+  return mo
+}
+
+function youtubeHandleAd (adContainer) {
```

Comment:
I think it'd make more sense to name this function to `youtubeMusicHandleAd` instead.